src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,746 @@
+              (hcievt->evtserial.evt.payload[2] == 0xFC)) {
+            phase_tx_power = true;
+            if (hcievt->evtserial.evt.payload[3] != 0) {
+#if defined(PRINT_IPCC_INFO)
+              printf("Error: wrong Tx power\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+            }
+            /* rx data is no more useful : not stored in the _rxbuff */
+            break;
+          }
+          /* check if the reset phase is in progress (opcode is 0x0C03) */
+          if ((hcievt->evtserial.evt.evtcode == TL_BLEEVT_CC_OPCODE) &&
+              (hcievt->evtserial.evt.payload[0] == 0x01) &&
+              (hcievt->evtserial.evt.payload[1] == 0x03) &&
+              (hcievt->evtserial.evt.payload[2] == 0x0C)) {
+            phase_reset = true;
+            if (hcievt->evtserial.evt.payload[3] != 0) {
+#if defined(PRINT_IPCC_INFO)
+              printf("Error: wrong reset\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+            }
+          }
+        }
+        __disable_irq();
+        /* store received data in the _rxbuff buffer */
+        len = hcievt->evtserial.evt.plen + TL_EVT_HDR_SIZE;
+        if (len <= BLE_MODULE_SHARED_MEM_BUFFER_SIZE - _write_index) {
+          /* at the position of the _write_index */
+          memcpy((uint8_t *)&_rxbuff[_write_index], (uint8_t *)&hcievt->evtserial, len);
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    case TL_ACL_DATA_PKT_TYPE: {
+        TL_AclDataSerial_t *acl = &(((TL_AclDataPacket_t *)hcievt)->AclDataSerial);
+        __disable_irq();
+        len = acl->length + 5;
+        if (len <= BLE_MODULE_SHARED_MEM_BUFFER_SIZE - _write_index) {
+          memcpy((uint8_t *)&_rxbuff[_write_index], (uint8_t *)acl, len);
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    default:
+      /* should not happen */
+#if defined(PRINT_IPCC_INFO)
+      printf("BLE TL evt_received, wrong type:%d\r\n", hcievt->evtserial.type);
+#endif /*(PRINT_IPCC_INFO)*/
+      while (1); /* let's block to check */
+      break;
+  }
+#if defined(PRINT_IPCC_INFO)
+  if (data_overflow) {
+    printf("Error: data read overflow\r\n");
+  }
+#endif /*(PRINT_IPCC_INFO)*/
+
+  /*  In case Event belongs to the Evt Pool we need to inform  */
+  if (((uint8_t *)hcievt >= EvtPool) && ((uint8_t *)hcievt < (EvtPool + POOL_SIZE))) {
+    /*  Free the message from shared memory */
+    TL_MM_EvtDone(hcievt);
+  }
+}
+
+/* to send BLE packet to the SharedMem */
+uint16_t mbox_write(uint8_t type, uint16_t len, const uint8_t *pData)
+{
+  TL_CmdPacket_t *bleCmdBuf = &BleCmdBuffer;
+  // Note: Until enum is available
+  // type 01 Command
+  // type 02 ACL DATA
+  // type 03 SCO Voice (not supported)
+  // type 04 event - uplink (not supported)
+
+  /*  TO DO : MANAGE ACL DATA CASE in separate buffer */
```

Comment:
Is comment still valid ? If different buffers are needed, then please create a separate buffer.

Reply:
the  uint16_t mbox_write functiuon is coming from mbed (including it's own comment)

Reply:
Maybe the comment is not valid in mbed either.
In any case, if there is no plan to do something, better remove the comment.

Reply:
ok